flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusReporter.java

```diff
@@ -130,7 +139,11 @@ public void notifyOfAddedMetric(final Metric metric, final String metricName, fi
 				metric.getClass().getName());
 			return;
 		}
-		collector.register();
+		try {
+			collector.register();
+		} catch (Exception e) {
+			LOG.warn("There was a problem registering metric {}: {}", metricName, e);
```

Comment:
We usually don't include placeholders for exceptions (because they are added implicitly), i.e. it should be ```LOG.warn("There was a problem registering metric {}.", metricName, e);```